qemu/tests/qemu_guest_agent.py

```diff
@@ -752,44 +753,137 @@ def get_allocation_bitmap():
+            """
+            os_type = params.get("os_type", "linux")
+            if os_type == "linux":
+                down_cmd = "ip link set %s" % ifname
+                if disabled:
+                    down_cmd += " down"
+                else:
+                    down_cmd += " up"
+                session.cmd(down_cmd)
+                time.sleep(10)
+            else:
+                if disabled:
+                    utils_net.disable_windows_guest_network(session, ifname)
+                else:
+                    utils_net.enable_windows_guest_network(session, ifname)
 
-        interfaces_after_add = self.gagent.get_network_interface()
+        session_serial = self.vm.wait_for_serial_login()
+        mac_addr = self.vm.get_mac_address()
+        error_context.context("Check if 'guest-network-get-interfaces' works",
+                              logging.info)
+        ret = self.gagent.get_network_interface()
+        interface_name, index_if = get_interface_attr(ret, mac_addr)
+        if not interface_name:
+            test.fail("Did not get the expected interface,"
+                      " the network info is \n%s." % ret)
 
-        bridge_list = [_ for _ in interfaces_after_add if _ not in
-                       interfaces_pre_add]
-        if (len(bridge_list) != 1) or \
-           ("lo_brige" != bridge_list[0]["name"]):
-            test.fail("the interface list info after interface was down "
-                      "was not as expected")
+        # check the available interface name is the same with guest.
+        if self.params.get("os_type") == "linux":
+            interface_name_guest = utils_net.get_linux_ifname(session_serial, mac_addr)
+        else:
+            interface_name_guest = utils_net.get_windows_nic_attribute(session_serial,
+                                                                       "macaddress",
+                                                                       mac_addr,
+                                                                       "netconnectionid")
+        if interface_name != interface_name_guest:
+            test.fail("Get the wrong interface name, value from qga is: %s; "
+                      "the expected is: %s" % (interface_name, interface_name_guest))
+
+        # check ip address is the same with guest
+        ip_addr_check(ret, index_if, interface_name)
+
+        # set down/up the interface
+        error_context.context("Set down the interface in guest.", logging.info)
+        manage_guest_nic(session_serial, interface_name, disabled=True)
+        ret_after_down = self.gagent.get_network_interface()
+        interface_name_down, index_if_down = get_interface_attr(ret_after_down,
+                                                                mac_addr)
+        if interface_name_down:
+            test.fail("From qga result that the interface is still enabled,"
+                      " detailed info is:\n %s" % ret_after_down)
```

Comment:
Currently, get_network_interface can get interface name after it is downed for linux guest,  Fow windows guest, It cann't get interface name.   I have sent a mail to developer to confirm this behavior.  Anyway,   linux and windows guest get  different output information after interface is downed.